### PR TITLE
Session Management API enhancements

### PR DESCRIPTION
**Pull request by jcgarciaa**
### Proposed changes in this pull request
Includes new features to the session management API as per wso2/product-is#12193:
- Search active sessions on the system: includes filtering (by user agent, client IP address and login time among other fields) and pagination capabilities.
- Retrieve a specific session for a certain user.

### Related PRs
1. Merge wso2/carbon-identity-framework#3959 [Take note of released version number].
2. Merge wso2/identity-api-user#143 [After updating carbon-identity-framework version on pom file. Take note of released version number]
3. Merge wso2/identity-rest-dispatcher#398 [After updating identity-api-user version on pom file. Take note of released version number]
4. **Merge this PR [After updating carbon-identity-framework, identity-api-user and identity-rest-dispatcher versions on pom file.]**
5. Merge wso2/docs-is#2874.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2075,7 +2075,7 @@
     <properties>
 
         <!--Carbon Identity Framework Version-->
-        <carbon.identity.framework.version>5.20.342</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.20.345-SNAPSHOT</carbon.identity.framework.version>
```
Comment (jcgarciaa) [inline pom.xml:2078]: Version number to be confirmed after merging wso2/carbon-identity-framework#3959

Comment (jcgarciaa) [inline pom.xml:2078]: Version number to be confirmed after merging wso2/carbon-identity-framework#3959

File: pom.xml [hunk 2]
```diff
@@ -2183,8 +2183,8 @@
         <identity.branding.preference.management.version>1.0.1</identity.branding.preference.management.version>
 
         <!-- Identity REST API feature -->
-        <identity.api.dispatcher.version>2.0.0</identity.api.dispatcher.version>
-        <identity.user.api.version>1.2.2</identity.user.api.version>
+        <identity.api.dispatcher.version>2.0.7-SNAPSHOT</identity.api.dispatcher.version>
+        <identity.user.api.version>1.2.3-SNAPSHOT</identity.user.api.version>
```
Comment (jcgarciaa) [inline pom.xml:2187]: Version number to be confirmed after merging wso2/identity-api-user#143

File: pom.xml (hunk 1 above)
Comment (ThaminduDilshan) [reply pom.xml]: bumped

File: pom.xml (hunk 2 above)
Comment (ThaminduDilshan) [reply pom.xml]: bumped

File: pom.xml [hunk 3]
```diff
@@ -2183,8 +2183,8 @@
         <identity.branding.preference.management.version>1.0.1</identity.branding.preference.management.version>
 
         <!-- Identity REST API feature -->
-        <identity.api.dispatcher.version>2.0.0</identity.api.dispatcher.version>
-        <identity.user.api.version>1.2.2</identity.user.api.version>
+        <identity.api.dispatcher.version>2.0.7-SNAPSHOT</identity.api.dispatcher.version>
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply pom.xml]: no need

File: modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml [hunk 4]
```diff
@@ -125,23 +125,24 @@
     </jaxrs:server>
     <jaxrs:server id="users" address="/users/v1">
         <jaxrs:serviceBeans>
+            <bean class="org.wso2.carbon.identity.rest.api.user.application.v1.MeApi"/>
```
Comment (piraveena) [inline modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml:128]: why we added application management apis?

File: modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml [hunk 5]
```diff
@@ -125,23 +125,24 @@
...
+            <bean class="org.wso2.carbon.identity.rest.api.user.backupcode.v1.MeApi"/>
             <bean class="org.wso2.carbon.identity.rest.api.user.challenge.v1.MeApi"/>
-            <bean class="org.wso2.carbon.identity.rest.api.user.approval.v1.MeApi"/>
-            <bean class="org.wso2.carbon.identity.rest.api.user.session.v1.MeApi"/>
+            <bean class="org.wso2.carbon.identity.rest.api.user.challenge.v1.UserIdApi"/>
             <bean class="org.wso2.carbon.identity.rest.api.user.fido2.v1.MeApi"/>
-            <bean class="org.wso2.carbon.identity.rest.api.user.totp.v1.MeApi"/>
-            <bean class="org.wso2.carbon.identity.rest.api.user.backupcode.v1.MeApi"/>
+            <bean class="org.wso2.carbon.identity.rest.api.user.functionality.v1.MeApi"/>
+            <bean class="org.wso2.carbon.identity.rest.api.user.functionality.v1.UserIdApi"/>
```
Comment (piraveena) [inline modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml:139]: why introduced these now? Didn't we have these ?

Comment (ThaminduDilshan) [reply modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml:139]: This is a alphabetic re-ordering

**Review by piraveena — APPROVED** [review]

File: modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml (hunk 4 above)
Comment (ThaminduDilshan) [reply modules/api-resources/api-resources-full/src/main/webapp/WEB-INF/beans.xml:128]: This is a alphabetic re-ordering

**Pull request merged**